frame/vault/src/lib.rs

```diff
@@ -971,10 +971,12 @@ pub mod pallet {
 						allocation
 							.mul_floor(<T::Convert as Convert<T::Balance, u128>>::convert(aum)),
 					);
-					if balance >= max_allowed {
+					if balance > max_allowed {
 						Ok(FundsAvailability::Depositable(balance - max_allowed))
-					} else {
+					} else if balance < max_allowed {
 						Ok(FundsAvailability::Withdrawable(max_allowed - balance))
+					} else {
+						Ok(FundsAvailability::None)
```

Comment:
In that case I would recommend to use `match`/`cmp` pattern. See [clippy's rule](https://rust-lang.github.io/rust-clippy/master/index.html#comparison_chain) for details.

Reply:
Thanks! Resolved!